### PR TITLE
Manually update istio/api

### PR DESCRIPTION
**Pull request by Monkeyanator**
isito/api automator PR never got created for [https://github.com/istio/api/pull/1934](https://github.com/istio/api/pull/1934) due to postsubmit [failure](https://prow.istio.io/view/gs/istio-prow/logs/update_api_dep_api_postsubmit/1378169589392740352), this manually updates istio/api

@douglas-reid for the code changes

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml [hunk 1]
```diff
@@ -1538,6 +1538,370 @@ spec:
                     local:
                       inline_string: "envoy.wasm.stats"
 ---
+# Source: istio-discovery/templates/telemetryv2_1.8.yaml
+# Note: metadata exchange filter is wasm enabled only in sidecars.
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:1542]: where did this come from?

File: pilot/pkg/model/telemetry.go [hunk 2]
```diff
@@ -142,7 +142,7 @@ func shallowMergeTracing(parent, child *tpb.Telemetry) *tpb.Telemetry {
 	}
 
 	// TODO: use wrapper in API to allow 0-valued override
-	if childTracing.GetRandomSamplingPercentage() != 0 {
+	if childTracing.RandomSamplingPercentage.GetValue() != 0 {
```
Comment (howardjohn) [inline pilot/pkg/model/telemetry.go:145]: I don't think this is what we want, instead we should have 
```suggestion
	if childTracing.RandomSamplingPercentage != nil {
```

Your code keeps the same behavior as today, but we changed the API so we could change the same behavior

**Comment by howardjohn** [comment]
the telemetry change can be fixed by telemetry team in a followup

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml (hunk 1 above)
Comment (Monkeyanator) [reply manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml:1542]: From the `make gen` step, not sure why it gets added

File: pilot/pkg/model/telemetry.go (hunk 2 above)
Comment (Monkeyanator) [reply pilot/pkg/model/telemetry.go]: Gotcha, fixed

**Pull request merged**